troubleshooting/known_issues/SBPS_boot_fail.md

```diff
@@ -0,0 +1,114 @@
+
+Issue can be identified with the symptoms below:
+
+Compute/UAN node (iSCSI Initiator) boot fails with the below messages in the console log:
+
+```text
+2024-11-12T20:07:50.17117 [  100.620313] dracut-pre-mount[3827]: iscsiadm: No active sessions.
+2024-11-12T20:13:00.44921 [  402.905538] dracut-pre-mount[5379]: ls: cannot access '/dev/mapper/594b7c79-9cb5-48f6-bceb-18f3596f0dbb_rootfs': No such file or directory
+2024-11-12T20:13:00.44925 [  402.924254] dracut-pre-mount[3814]: Warning: sbps-add-content.sh failed.
+2024-11-12T20:13:00.44926 [  402.952550] dracut-pre-mount[3809]: Warning: Unable to prepare squashfs file /tmp/cps/rootfs, dropping to debug.
+```
+
+SBPS Marshal agent crashes with the below messages on the worker node (iSCSI Target):
+
+```bash
+ncn-w002:~ # systemctl status sbps-marshal.service
+× sbps-marshal.service - System service that manages Squashfs images projected via iSCSI for IMS, PE, and other ancillary images similar to PE.
+     Loaded: loaded (/usr/lib/systemd/system/sbps-marshal.service; enabled; preset: disabled)
+     Active: failed (Result: exit-code) since Tue 2024-11-12 19:20:05 UTC; 18h ago
+   Duration: 2.232s
+    Process: 1239406 ExecStart=/usr/lib/sbps-marshal/bin/sbps-marshal (code=exited, status=1/FAILURE)
+   Main PID: 1239406 (code=exited, status=1/FAILURE)
+        CPU: 1.394s
+
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]:   File "/usr/lib/sbps-marshal/bin/sbps-marshal", line 8, in <module>
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]:     sys.exit(main())
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]:              ^^^^^^
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]:   File "/usr/lib/sbps-marshal/lib/python3.12/site-packages/bin/agent.py", line 270, in main
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]:     rootfs_s3_etag = artifact["link"]["etag"]
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]:                      ~~~~~~~~~~~~~~~~^^^^^^^^
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]: KeyError: 'etag'
+Nov 12 19:20:05 ncn-w002 systemd[1]: sbps-marshal.service: Main process exited, code=exited, status=1/FAILURE
+Nov 12 19:20:05 ncn-w002 systemd[1]: sbps-marshal.service: Failed with result 'exit-code'.
+Nov 12 19:20:05 ncn-w002 systemd[1]: sbps-marshal.service: Consumed 1.394s CPU time.
+```
+
+The complete log for the SBPS Marshal agent `systemd` service can be obtained from `journalctl` log as below:
+
+```bash
+    # journalctl -u sbps-marshal.service
+```
+
+## Workaround Description
+
+Step 1: Check whether the image to be projected has `etag` in the image manifest
+
+example #1:
+
+```bash
+# cray ims images describe 3f89b76d-7578-4ac3-8891-0f8b4335b06d
+arch = "x86_64" 
+created = "2024-08-15T14:38:35.897457+00:00" 
+id = "3f89b76d-7578-4ac3-8891-0f8b4335b06d" 
+name = "kpreslan-20240815093713" 
+
+[link] etag = ""  <---------------------- `etag` set to NULL
+path = "s3://boot-images/3f89b76d-7578-4ac3-8891-0f8b4335b06d/manifest.json" 
+type = "s3" 
+
+[metadata]

```

Comment:
This is not the problem, it's the missing "etag" in the manifest.json that causes the issue.

The first step is to provide the step to scan the system to identify whether or not there are any problematic manifest.json file.  Look in the description in the ticket, I added the information on how to identify which images have the issue.  Once you've identified the images, then describe how to fix them.

Reply:
@dborman-hpe Thanks for that clarification!.  I looked at the ticket, so do we want suggest the step to add missing 'etag' to the manifest or delete the offending manifest ?

Reply:
As you mentioned in the note, if we are suggesting the add missing 'etag' then we need to provide steps to add etag and what value of etag.

Reply:
Ok I see Jason has added comment to add etag instead of deleting the images. But need exact steps/commands for the same.